bedrock/settings/__init__.py

```diff
@@ -156,20 +156,22 @@ def _override_csp(csp, append: dict[str, list[str]] = None, replace: dict[str, l
 # /cms-admin/images/ loads just-uploaded images as blobs.
 CMS_ADMIN_IMAGES_CSP = _override_csp(CONTENT_SECURITY_POLICY, append={"img-src": ["blob:"]})
 
+# The CMS admin frames itself for page previews.
+CMS_ADMIN_CSP = _override_csp(CONTENT_SECURITY_POLICY, replace={"frame-ancestors": [csp.constants.SELF]})
+
 CSP_PATH_OVERRIDES = {
     # Order them from most specific to least.
     "/cms-admin/images/": CMS_ADMIN_IMAGES_CSP,
+    "/cms-admin/": CMS_ADMIN_CSP,
 }
 
 if csp_ro_report_uri:
     # Path based overrides for report-only CSP.
-    CMS_ADMIN_CSP_RO = _override_csp(CONTENT_SECURITY_POLICY_REPORT_ONLY, replace={"frame-ancestors": [csp.constants.SELF]})
     CMS_ADMIN_IMAGES_CSP_RO = _override_csp(CONTENT_SECURITY_POLICY_REPORT_ONLY, append={"img-src": ["blob:"]})
 
     CSP_PATH_OVERRIDES_REPORT_ONLY = {
         # Order them from most specific to least.
         "/cms-admin/images/": CMS_ADMIN_IMAGES_CSP_RO,
-        "/cms-admin/": CMS_ADMIN_CSP_RO,
     }
```

Comment:
Q out of ignorance: the admin images seem to be duplicated in both overrides — is that necessary due to the way the overrides are deepcopied and in which order, as no matter what is in enforced path overrides, that doesn't automagically translate into the equivalent RO definition at the moment?

Reply:
Yeah, we set the enforced CSP. Then deepcopy it to create the report-only CSP. Then the overrides use each of those as a base.

As a tree this looks like:
- CSP
  - CSP_RO
    - CSP_RO_OVERRIDES
  - CSP_OVERRIDES

Reply:
So basically the path override rules should not be moved, but duplicated — not to disappear from the RO policy, sending false violations for the admin paths?

(Or, instead, the CSP_RO_OVERRIDES should start with a copy from CSP_OVERRIDES for consistency/predictability?)

Reply:
IIUC if the current model stays, the removed CMS_ADMIN_CSP_RO here above need to be kept, to mirror what's being added to CMS_ADMIN_CSP, not to start triggering RO violations.

Reply:
Yep, nice catch, thank you. That's fixed now. I also tried to move code together so it's a little easier to see what's going on.